sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,1033 @@
+              } else if (updateResponse.getResult() == DocWriteResponse.Result.NOOP) {
+                promise.success(false);
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  falied for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              }
+
+              long stopTime = System.currentTimeMillis();
+              long elapsedTime = stopTime - startTime;
+              ProjectLogger.log(
+                  "ElasticSearchUtilRest updateData method end at =="
+                      + stopTime
+                      + " for Type "
+                      + type
+                      + " ,Total time elapsed = "
+                      + elapsedTime,
+                  LoggerEnum.PERF_LOG);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              ProjectLogger.log(
+                  "ElasticSearchUtilRest : updateData exception occured:" + e.getMessage(),
+                  LoggerEnum.ERROR.name());
+              promise.failure(e);
+            }
+          };
+      ConnectionManager.getRestClient().updateAsync(updateRequest, listener);
+
+    } else {
+      ProjectLogger.log(
+          "ElasticSearchUtilRest : updateData Requested data is invalid.", LoggerEnum.INFO.name());
+    }
+    return promise.future();
+  }
+
+  /**
+   * This method will provide data form ES based on incoming identifier. we can get data by passing
+   * index and identifier values , or all the three
+   *
+   * @param type String
+   * @param identifier String
+   * @return Map<String,Object> or null
+   */
+  @Override
+  public Future<Map<String, Object>> getDataByIdentifier(
+      String index, String type, String identifier) {
+    long startTime = System.currentTimeMillis();
+    Promise<Map<String, Object>> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchUtilRest getDataByIdentifier method started at =="
+            + startTime
+            + " for Type "
+            + type,
+        LoggerEnum.PERF_LOG);
+
+    Map<String, String> mappedIndexAndType = ElasticSearchHelper.getMappedIndexAndType(index, type);
+
+    GetRequest getRequest =
+        new GetRequest(
+            mappedIndexAndType.get(JsonKey.INDEX),
+            mappedIndexAndType.get(JsonKey.TYPE),
+            identifier);
+
+    ActionListener<GetResponse> listener =
+        new ActionListener<GetResponse>() {
+          @Override
+          public void onResponse(GetResponse getResponse) {
+            if (getResponse.isExists()) {
+              Map<String, Object> sourceAsMap = getResponse.getSourceAsMap();
+              if (MapUtils.isNotEmpty(sourceAsMap)) {
+                promise.success(sourceAsMap);
+                long stopTime = System.currentTimeMillis();
```

Comment:
use pre define method to calculate method execution end time.

Reply:
Done